### PR TITLE
Minor code changes

### PR DESCRIPTION
**Pull request by ovalseven8**

### CODE REVIEW

File: src/main/java/org/saltyrtc/client/cookie/Cookie.java [hunk 1]
```diff
@@ -14,9 +14,6 @@
 
 /**
  * A SaltyRTC cookie.
- *
- * This data structure is immutable, but it can be cloned
- * using a copy constructor.
```
Comment (dbrgn) [inline src/main/java/org/saltyrtc/client/cookie/Cookie.java]: Why did you remove this?

File: src/main/java/org/saltyrtc/client/keystore/Box.java [hunk 2]
```diff
@@ -86,4 +86,11 @@ public boolean equals(Object o) {
         return Arrays.equals(this.data, other.getData())
             && Arrays.equals(this.nonce, other.getNonce());
     }
+
+    @Override
+    public int hashCode() {
+        int result = Arrays.hashCode(nonce);
+        result = 31 * result + Arrays.hashCode(data);
+        return result;
+    }
```
Comment (dbrgn) [inline src/main/java/org/saltyrtc/client/keystore/Box.java]: I think `Box` is just an internal API and it doesn't need to be hashable. What's the advantage of overriding `hashCode` in this case? Also, what's the `31` magic number?

File: src/main/java/org/saltyrtc/client/signaling/InitiatorSignaling.java [hunk 3]
```diff
@@ -599,7 +599,7 @@ Peer getPeerWithId(short id) throws SignalingException {
             return this.server;
         } else if (this.isResponderId(id)) {
             //noinspection ConstantConditions
-            if (this.getState() == SignalingState.TASK && this.responder != null & this.responder.getId() == id) {
+            if (this.getState() == SignalingState.TASK && this.responder != null && this.responder.getId() == id) {
```
Comment (dbrgn) [inline src/main/java/org/saltyrtc/client/signaling/InitiatorSignaling.java]: Oh, that's a nice catch! Especially since the noinspection comment above silences the warning in IntelliJ :face_with_head_bandage: Fortunately it doesn't change anything about the logic since both sides are booleans.

Can you also remove the noinspection comment above?

File: src/main/java/org/saltyrtc/client/helpers/RandomHelper.java [hunk 4]
```diff
@@ -13,14 +13,17 @@
 
 public class RandomHelper {
 
+    private static Random random = new Random();
+    private static SecureRandom secureRandom = new SecureRandom();
```
Comment (dbrgn) [inline src/main/java/org/saltyrtc/client/helpers/RandomHelper.java]: Note: Re-using the RNG *might* have security implications, but it should be safe in this case as far as I can tell after some research.

File: src/main/java/org/saltyrtc/client/cookie/Cookie.java (hunk 1 above)
Comment (ovalseven8) [reply src/main/java/org/saltyrtc/client/cookie/Cookie.java]: This shouldn't be possible with an immutable class, I believe.

```
class Scratch {
    public static void main(String[] args) {
        Cookie cookie = new Cookie();
        byte[] cookieBytes = cookie.getBytes();
        System.out.println(Arrays.toString(cookieBytes));
        cookieBytes[0] = 54;
        System.out.println(Arrays.toString(cookie.getBytes()));
    }
}
```

Output:
```
[14, -97, 28, 37, -35, -33, 64, 59, 29, -36, 47, -118, 65, 68, 1, -47]
[54, -97, 28, 37, -35, -33, 64, 59, 29, -36, 47, -118, 65, 68, 1, -47]
```

File: src/main/java/org/saltyrtc/client/keystore/Box.java (hunk 2 above)
Comment (ovalseven8) [reply src/main/java/org/saltyrtc/client/keystore/Box.java]: You're right. It does not hurt though.

It's good practice to implement the `hashCode()` method when there's also an implemented `equals()` method.

This code is generated by the IDE. The `31` is just a popular number to use: https://stackoverflow.com/a/299748

File: src/main/java/org/saltyrtc/client/cookie/Cookie.java (hunk 1 above)
Comment (dbrgn) [reply src/main/java/org/saltyrtc/client/cookie/Cookie.java]: Yeah, that's a good point. Unfortunately Java does not give us write protection the way Rust does.

I guess it's ok if we simply remove the note. From the API it's already obvious that the values shouldn't be modified, I hope.

File: src/main/java/org/saltyrtc/client/helpers/RandomHelper.java (hunk 4 above)
Comment (ovalseven8) [reply src/main/java/org/saltyrtc/client/helpers/RandomHelper.java]: So, should I keep it as before or is this change ok?

File: src/main/java/org/saltyrtc/client/signaling/InitiatorSignaling.java [hunk 5]
```diff
@@ -598,8 +598,7 @@ Peer getPeerWithId(short id) throws SignalingException {
         if (id == SALTYRTC_ADDR_SERVER) {
             return this.server;
         } else if (this.isResponderId(id)) {
-            //noinspection ConstantConditions
-            if (this.getState() == SignalingState.TASK && this.responder != null && this.responder.getId() == id) {
+            if ((this.getState() == SignalingState.TASK) && (this.responder != null) && (this.responder.getId() == id)) {
```
Comment (dbrgn) [inline src/main/java/org/saltyrtc/client/signaling/InitiatorSignaling.java]: Can you remove the parentheses? They add unnecessary noise.

File: src/main/java/org/saltyrtc/client/helpers/RandomHelper.java (hunk 4 above)
Comment (dbrgn) [reply src/main/java/org/saltyrtc/client/helpers/RandomHelper.java]: I think it's OK. It might improve performance slightly depending on the platform.

File: src/main/java/org/saltyrtc/client/signaling/InitiatorSignaling.java (hunk 5 above)
Comment (ovalseven8) [reply src/main/java/org/saltyrtc/client/signaling/InitiatorSignaling.java]: Ups, I like those parentheses because you don't have to think about the operation order.

Will change though. :smile:

**Comment by ovalseven8** [comment]
Done. Should I rebase?

**Comment by dbrgn** [comment]
@ovalseven8 no, fixup commits are fine :slightly_smiling_face: We usually squash-merge PRs anyways.
